### ISSUE TITLE
Problem with Microsoft.Extensions.Options when creating a NetCore 2.0 app

### ISSUE THREAD

**Issue by Roblinde** [contentful/contentful.net#124]
See this pull request for more information #122 

![image](https://user-images.githubusercontent.com/1835323/31708615-ca3226c8-b3ef-11e7-8bb1-692c484ca3a0.png)

**Comment by Roblinde** [comment]
@claylaut Did you do anything in particular when creating your new application or does this happen directly when running a new AspNetCore app?

**Comment by claylaut** [comment]
@Roblinde, I didn't try on a new application. This currently happing on my actual application when registering contentful with the below configuration.
```
  "ContentfulOptions": {
    "DeliveryApiKey": "xxxxx",
    "ManagementApiKey": "xxxxxx",
    "SpaceId": "xxxxxx",
    "UsePreviewApi": false,
    "MaxNumberOfRateLimitRetries": 0
  }
```

**Comment by Roblinde** [comment]
I see, did this application upgrade from .NET Core 1.x to 2.0 or was it created as a 2.0?

Do you have a reference to the Microsoft.AspNetCore.All package?

Are you using a global.json file to pin an SDK version?

If you create a new blank AspNetCore 2.0 application and reference Contentful, does that work?

**Comment by claylaut** [comment]
This application is an upgrade from .NET Core 1.x to 2.0.
Yes, I m referencing `Microsoft.AspNetCore.All`.
`ContentfulOptions` are set in `appsettings.json`
I didn't try a new application but I've referenced the forked version of `Contentful` with the `.NET Core 2` and it worked during runtime.

**Comment by Roblinde** [comment]
Very strange. It works directly for me when creating a new ASP.NET Core 2.0 app or upgrading one from 1.1.

Which version of the SDK are you running?

Could you perhaps paste a screenshot of your NuGet packages, perhaps I can spot something there.

**Comment by claylaut** [comment]
We had reference to`contentful.csharp v1.5.0` and another project was using `contentful.aspnetcore v1.5.0`.
Currently, I m still working against the forked version since I had to continue the migration to `.NET CORE 2.0`.

* In your configuration file you have multiple configurations?
* Are you injecting the configuration through DI?

![alt text](https://i.imgur.com/VDk6ZO5.jpg "error")

**Comment by Roblinde** [comment]
Yes, I have both a `Logging` and a `ContentfulOptions` section. And I am injecting through the extension method. There's one difference though. In AspNetCore 2.0 apps they've move over to `IConfiguration` instead of `IConfigurationRoot`. It shouldn't make much of a difference, but could be something to look into.

**Comment by claylaut** [comment]
I will try to switch to `IConfiguration`, Can you please upload your sample `.NET CORE 2` Project ? so I can use it as a reference. 

I will try to start using `1.5.1` again soon.

**Comment by Roblinde** [comment]
Sure, redacted the access tokens. But otherwise everything is as it should be. Providing both the upgraded project (40) and the one that was 2.0 directly (39).

Hope you find something to help you out.

[projects.zip](https://github.com/contentful/contentful.net/files/1398155/projects.zip)

**Comment by claylaut** [comment]
Cheers, I will keep you updated once I've tested it.

**Comment by claylaut** [comment]
I've managed to make it work with `contentful.aspnetcore v1.5.1`.
I had another 3rd party nuget package which contained an old reference of the .net core and it causing an issue on the configuration. Unfortunately, it was throwing an exception in the Contentful library, which mislead me. I've updated the other library and now contentful is working fine perfectly.

Thanks for your fast response and support.
